contracts/oracle/VoltSystemOracle.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {Constants} from "./../Constants.sol";
+import {Initializable} from "@openzeppelin/contracts/proxy/utils/Initializable.sol";
+import {IVoltSystemOracle} from "./IVoltSystemOracle.sol";
+
+/// @notice contract that receives a fixed interest rate upon construction,
+/// and then linearly interpolates that rate over a 1 year period into the VOLT price
+/// after the oracle start time.
+/// Interest can compound annually. Assumption is that this oracle will only be used until
+/// Volt 2.0 ships. Maximum amount of compounding periods on this contract at 2% APR
+/// is 6192 years, which is more than enough for this use case.
```

Comment:
code4rena: overflows cause revert if oraclePrice is greater than uint max, or getCurrentOraclePrice multiplication goes over uint max.